File: Modules/ServerTypes/Ticket/CommentColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { CommentRow } from "./CommentRow";

export interface CommentColumns {
    Id: Column<CommentRow>;
    Comment: Column<CommentRow>;
    TicketTitle: Column<CommentRow>;
    Username: Column<CommentRow>;
    DateCreated: Column<CommentRow>;
}

export class CommentColumns extends ColumnsBase<CommentRow> {
    static readonly columnsKey = 'Ticket.Comment';
    static readonly Fields = fieldsProxy<CommentColumns>();
}

File: Modules/Ticket/Comment/CommentColumns.cs
namespace Ticketing.Ticket.Columns;

[ColumnsScript("Ticket.Comment")]
[BasedOnRow(typeof(CommentRow), CheckNames = true)]
public class CommentColumns
{
    [EditLink, Width(250)]
    public string Comment { get; set; }
    [Width(125)]
    public string Username { get; set; }
    [Width(125)]
    public DateTime DateCreated { get; set; }
}